tests/bsim/bluetooth/host/central/src/main.c

```diff
@@ -70,6 +72,21 @@ static void test_central_connect_timeout_with_timeout(uint32_t timeout_ms)
 	err = bt_conn_le_create(&peer, &create_param, BT_LE_CONN_PARAM_DEFAULT, &conn);
 	TEST_ASSERT(err == 0, "Failed starting initiator (err %d)", err);
 
+	if (stack_load) {
+		/* Claim all the buffers so that the stack cannot handle the timeout */
+		for (int i = 0; i < CONFIG_BT_BUF_CMD_TX_COUNT; i++) {
+			bufs[i] = bt_hci_cmd_create(BT_HCI_LE_ADV_ENABLE, 0);
+			TEST_ASSERT(bufs[i] != NULL, "Failed to claim all command buffers");
+		}
+		/* Hold all the buffers until after we expect the connection to timeout */
+		err = k_sem_take(&sem_failed_to_connect, K_MSEC(expected_conn_timeout_ms + 50));
+		TEST_ASSERT(err == -EAGAIN, "Callback ran with no buffers available", err);
+		/* Release all the buffers back to the stack */
+		for (int i = 0; i < CONFIG_BT_BUF_CMD_TX_COUNT; i++) {
+			net_buf_unref(bufs[i]);
+		}
```

Comment:
As applications are allowed to allocate command buffers, are we ready to support users reporting live-lock issues, and explaining/educating/debugging their applications when failing on such a live-lock?

Reply:
This test is verifying a valid behavior by releasing the command buffers. How is the case when application leaks/holds command buffers, are we not making it difficult to detect a system failure in this PR?  In a resource constraint device, an immediate failure can recover the system quicker (LOG_ERR/LOG_WRN is not going to help here though).

Reply:
What system failure are we making harder to detect? An application grabbing all the buffers and never releasing them? That will be obvious almost immediately when suddenly every Bluetooth API call starts failing.
The condition tested here can happen in normal operation when the application is doing everything right, but is unlucky with timings.
Not testing the case is asking for the NULL de-reference to re-appear in the future.

Reply:
> What system failure are we making harder to detect? An application grabbing all the buffers and never releasing them? That will be obvious almost immediately when suddenly every Bluetooth API call starts failing.

`net_buf_alloc` call is blocking with `K_FOREVER` in `bt_hci_cmd_create`, are you sure every Bluetooth API call will fail? A deadlock, and now a live-lock, both will not be detected, and hence preventing a system recovery.